### ISSUE TITLE
Unstable behavior of server 

### ISSUE THREAD

**Issue by mstfdkmn** [irods/irods_client_http_api#186]
## Bug Report

I had been testing the write operation of the data_objects endpoint. At some point I saw some strange behaviors:

- Looked like the server got stuck,
- It didn't response anything, even to the authentication request,
- Client operations hanged.
- The only thing I had to do was kill the process (restart the server).

Here is an example of what I encountered:

```
[irods@dev-u0137480 go-http-client]$ time curl -H "Authorization: Bearer $token" 'http://localhost:9000/irods-http-api/0.1.0/data-objects' -F  'op=write' -F 'lpath=/u0137480/home/rods/myfile.txt' -F 'count=125000' -F 'bytes=abcdfg'
^C

real    0m16.016s
user    0m0.006s
sys     0m0.004s
[irods@dev-u0137480 go-http-client]$ token=$(curl -X POST -su rods:hunter2 'http://localhost:9000/irods-http-api/0.1.0/authenticate')
^C
```

logs (I post older ones too):
```
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Header: (Host, localhost:9000)
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Header: (User-Agent, curl/7.61.1)
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Header: (Accept, */*)
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Header: (Authorization, Bearer 57136313-b9ac-4c20-a127-01e95b7a2343)
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Header: (Content-Length, 466)
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Header: (Content-Type, multipart/form-data; boundary=------------------------b8b5044cece33b29)
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Method: POST
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Version: 11
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Target: /irods-http-api/0.1.0/data-objects
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Keep Alive: true
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Has Content Length: true
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Chunked: false
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] on_read: Needs EOF: false
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] PARAM = (name=boundary, value=------------------------b8b5044cece33b29)
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] parse_multipart_form_data: CAPTURED CONTENT SIZE = [5]
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] parse_multipart_form_data: CAPTURED CONTENT SIZE = [30]
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] parse_multipart_form_data: CAPTURED CONTENT SIZE = [6]
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] parse_multipart_form_data: CAPTURED CONTENT SIZE = [6]
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] resolve_client_identity: Authorization value: [Bearer 57136313-b9ac-4c20-a127-01e95b7a2343]
http-api stdout | [2023-12-22 15:24:25.273] [P:571486] [debug] [T:571537] resolve_client_identity: Bearer token: [57136313-b9ac-4c20-a127-01e95b7a2343]

http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Header: (Host, localhost:9000)
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Header: (Authorization, Basic cm9kczpodW50ZXIy)
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Header: (User-Agent, curl/7.61.1)
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Header: (Accept, */*)
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Method: POST
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Version: 11
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Target: /irods-http-api/0.1.0/authenticate
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Keep Alive: true
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Has Content Length: false
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Chunked: false
http-api stdout | [2023-12-22 15:24:54.022] [P:571486] [debug] [T:571486] on_read: Needs EOF: false



http-api stdout | [2023-12-22 15:26:34.170] [P:571486] [debug] [T:571537] Evicted bearer token [7ff2dc9d-cdd8-4671-845b-b4d8dcfba47a].
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Header: (Host, localhost:9000)
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Header: (Authorization, Basic cm9kczpodW50ZXIy)
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Header: (User-Agent, curl/7.61.1)
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Header: (Accept, */*)
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Method: POST
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Version: 11
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Target: /irods-http-api/0.1.0/authenticate
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Keep Alive: true
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Has Content Length: false
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Chunked: false
http-api stdout | [2023-12-22 15:27:25.647] [P:571486] [debug] [T:571486] on_read: Needs EOF: false
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Header: (Host, localhost:9000)
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Header: (Authorization, Basic cm9kczpodW50ZXIy)
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Header: (User-Agent, curl/7.61.1)
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Header: (Accept, */*)
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Method: POST
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Version: 11
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Target: /irods-http-api/0.1.0/authenticate
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Keep Alive: true
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Has Content Length: false
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Chunked: false
http-api stdout | [2023-12-22 15:29:22.030] [P:571486] [debug] [T:571536] on_read: Needs EOF: false
2023-12-22 15:30:08,977 WARN exited: http-api (terminated by SIGKILL; not expected)

```

**Comment by korydraughn** [comment]
The only thing that stands out here is the mismatch between the `count` param and the `bytes` param.

As you already know, the HTTP API does not verify those values align. It just trusts the client passed the correct values. I feel that is probably the reason for the weird behavior _(or part of it)_.

We'll get that fixed.

**Comment by trel** [comment]
Feels like we just need to compare the bytes and count at the start of the operation...

If count isn't defined (or also possibly if count=0)... then perhaps we're in streaming mode.

If they are present and don't match, error immediately.

**Comment by korydraughn** [comment]
I think this is resolved by #194.

With that resolved, there's no known way to write more bytes than what's available.

**Comment by trel** [comment]
I think we can close this for now.  If we see unstableness/inconsistency again (for whatever reason)... we'll open a new issue.
